### PR TITLE
Remove useless call to addRoute for multilanguages shops

### PR DESCRIPTION
**Pull request by jocel1**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Performance : remove useless call to addRoute for multilanguages shops
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | verify urls are still working properly

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8209)

### CODE REVIEW

**Comment by jocel1** [comment]
https://blackfire.io/profiles/compare/6d35a416-f4ad-4ea6-97f1-25b411a0eb4e/graph

**Comment by aleeks** [comment]
Hello @jocel1 
It's not really a bug fix so i move this PR on `develop`branch, you need to rebase from develop!
Thank you!

**Review by Quetzacoalt91 — DISMISSED** [review]

**Comment by eternoendless** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[classes/Dispatcher.php, line 557 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8209#-KvSY1-mbv_FHXsNsRUI:-KvSY1-mbv_FHXsNsRUJ:b-6sa6ob) ([raw file](https://github.com/prestashop/prestashop/blob/378e2885f89214d686ee9dc39f5b0daf2e32f53e/classes/Dispatcher.php#L557)):*
> ```PHP
>      *
>      * @return array
>      */
> ```

Would you mind adding a description for this method and all of its parameters?

---

*[classes/Dispatcher.php, line 564 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8209#-KvSY9ujFw8_npAkXQdb:-KvSY9ujFw8_npAkXQdc:b-699ptf) ([raw file](https://github.com/prestashop/prestashop/blob/378e2885f89214d686ee9dc39f5b0daf2e32f53e/classes/Dispatcher.php#L564)):*
> ```PHP
>             $transform_keywords = array();
>             preg_match_all('#\\\{(([^{}]*)\\\:)?(' .
>                 implode('|', array_keys($keywords)) . ')(\\\:([^{}]*))?\\\}#', $regexp, $m);
> ```

Could you please break this line so we have one argument per line?

---

*[classes/Dispatcher.php, line 584 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8209#-KvXPeRxhS1NH5bCGLqA:-KvXPeRxhS1NH5bCGLqB:b-75c0yh) ([raw file](https://github.com/prestashop/prestashop/blob/378e2885f89214d686ee9dc39f5b0daf2e32f53e/classes/Dispatcher.php#L584)):*
> ```PHP
>                     $regexp = str_replace($m[0][$i], $prepend_regexp .
>                         '(?P<' . $keywords[$keyword]['param'] . '>' . $keywords[$keyword]['regexp'] . ')' .
>                         $append_regexp, $regexp);
> ```

Here too, one argument per line please

---

*[classes/Dispatcher.php, line 588 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8209#-KvXPgqSINpSy9noSEOu:-KvXPgqSINpSy9noSEOv:b-fulkv0) ([raw file](https://github.com/prestashop/prestashop/blob/378e2885f89214d686ee9dc39f5b0daf2e32f53e/classes/Dispatcher.php#L588)):*
> ```PHP
>                     $regexp = str_replace($m[0][$i], $prepend_regexp .
>                         '(' . $keywords[$keyword]['regexp'] . ')' .
>                         $append_regexp, $regexp);
> ```

Also here

---

*[classes/Dispatcher.php, line 616 at r1](https://reviewable.io:443/reviews/prestashop/prestashop/8209#-KvXPzzXofu1Q6kmPreb:-KvXPzzXofu1Q6kmPrec:b-adbiy0) ([raw file](https://github.com/prestashop/prestashop/blob/378e2885f89214d686ee9dc39f5b0daf2e32f53e/classes/Dispatcher.php#L616)):*
Per PSR-2:

> Argument lists MAY be split across multiple lines, where each subsequent line is indented once. When doing so, the first item in the list MUST be on the next line, and there MUST be only one argument per line.

So please put one argument per line :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8209)*

File: classes/Dispatcher.php [hunk 1]
```diff
@@ -681,7 +740,8 @@ public function validateRoute($route_id, $rule, &$errors = array())
      * @param bool $use_routes If false, don't use to create this url
      * @param string $anchor Optional anchor to add at the end of this url
      */
-    public function createUrl($route_id, $id_lang = null, array $params = array(), $force_routes = false, $anchor = '', $id_shop = null)
+    public function createUrl($route_id, $id_lang = null, array $params = array(), $force_routes = false, $anchor = '',
```
Comment (LittleBigDev) [inline classes/Dispatcher.php]: Care about IED's autoformat.
Sometimes it doesn't respect PSR (for instance, here there is a problem with [multi-line method arguments](http://www.php-fig.org/psr/psr-2/#44-method-arguments))

**Review by LittleBigDev — CHANGES_REQUESTED** [review]

File: classes/Dispatcher.php [hunk 2]
```diff
@@ -843,7 +858,8 @@ public function getController($id_shop = null)
 
         $controller = Tools::getValue('controller');
 
-        if (isset($controller) && is_string($controller) && preg_match('/^([0-9a-z_-]+)\?(.*)=(.*)$/Ui', $controller, $m)) {
+        if (isset($controller) && is_string($controller)
+            && preg_match('/^([0-9a-z_-]+)\?(.*)=(.*)$/Ui', $controller, $m)) {
```
Comment (LittleBigDev) [inline classes/Dispatcher.php]: The closing `) {` should be on its own line

File: classes/Dispatcher.php [hunk 3]
```diff
@@ -843,7 +858,8 @@ public function getController($id_shop = null)
 
         $controller = Tools::getValue('controller');
 
-        if (isset($controller) && is_string($controller) && preg_match('/^([0-9a-z_-]+)\?(.*)=(.*)$/Ui', $controller, $m)) {
+        if (isset($controller) && is_string($controller)
```
Comment (LittleBigDev) [inline classes/Dispatcher.php]: one statement per line

**Review by LittleBigDev — CHANGES_REQUESTED** [review]

**Review by LittleBigDev — APPROVED** [review]

**Comment by eternoendless** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8209#-:-KwdNk7fQrc4EGNOVIUV:bnfp4nl)*

**Comment by LittleBigDev** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r5.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8209#-:-KwdOJ2OiGfmqW2_a3JM:bnfp4nl)*

**Comment by eternoendless** [comment]
Thank you @jocel1

**Pull request merged**